parse/node.go

```diff
@@ -81,10 +82,42 @@ func (n *Node) Validate() error {
 
 		return fmt.Errorf("%s: too many keys", n.Pos())
 	}
-
+	if err := validateName(n.Name()); err != nil {
+		return fmt.Errorf("%s: %s", n.Pos(), err)
+	}
 	return n.setValues()
 }
 
+func validateName(name string) error {
+	var hasSpaces bool
+	var hasNonAlphaNum bool
+	var invalidRunes []rune
+	for _, letter := range []rune(name) {
+		if unicode.IsSpace(letter) {
+			hasSpaces = true
+		}
+		if !(unicode.IsLetter(letter) || unicode.IsNumber(letter) || letter == '-' || letter == '_' || letter == '.') {
+			hasNonAlphaNum = true
+			invalidRunes = append(invalidRunes, letter)
+		}
+	}
+	if hasSpaces {
+		return errors.New("resource name may not contain spaces")
```

Comment:
why not return this immediately instead of keeping track of another value and parsing the entire name?